bindgen-cli/options.rs

```diff
@@ -1084,3 +1136,28 @@ where
 
     Ok((builder, output, verbose))
 }
+
+/// Parse a [`CTypeKind`] from the given command line `input`.
+fn parse_c_type(input: &str) -> Result<CTypeKind, String> {
```

Comment:
is this really necessary? It might sound lazy but to be honest it might be easier to just let the user type whatever they want as a type and propagate those strings until they are written to the wrapper code file. This also gives you the "advantage" of supporting any type the user wants as long as it is defined in the headers.

Reply:
I am actually a bit stuck on this. I am trying to support calling these macros with C types, which can't be referenced from build scripts:
```c
typedef struct {
    int inner;
} my_type;

#define ACCESS_INNER(value) value.inner
```

I want to be able to specify that the `ACCESS_INNER` macro takes a `my_type`. To do that I was trying to switch everything over to just accept strings, but in `Function::parse` I need to generate a `Function` object which eventually needs a `TypeKind` for the return type and the argument types.

So my question is, is there a way to construct a `Function` that doesn't require me to parse the types, so I can just pass the strings through?